### ISSUE TITLE
Notmuch search result can't be displayed in neomutt...

### ISSUE THREAD

**Issue by kuochang** [LukeSmithxyz/mutt-wizard#629]
Hello, I am a newbie to Linux and I've learned a lot from Luke through his YouTube videos. I used mutt wizard to backup all my 60K emails from O365 and it works great in the beginning...

Recently one strange thing happened. When I use notmuch search(press ctl-f) to look for certain emails, it displays nothing after October 22nd. Meaning that it shows part of emails that match my notmuch criteria and displays only those emails before October 22nd.

However, when I do notmuch search in command line(notmuch search + criteria), the terminal displays all matched emails, even after October 22nd. In other words, notmuch search works perfectly in command line but it displays part of the results(some emails are missing) in neomutt...

Does anyone have the similar experience? How should I fix this?

Please let me know any further info you need to investigate in my case? Thanks in advance!
-Charles

**Comment by SentToDevNull** [comment]
@kuochang I believe Office365 disabled "basic" authentication late last October. (That week of October was the exact same time it stopped working for me.) To use the "modern" authentication with Office365, you'll need to authenticate using oauth through Exchange. This isn't built into `neomutt` (but is built into `mutt` as of last August; sadly, mutt-wizard only works with `neomutt`, not `mutt`. 

To get it to work, use `davmail`, a MS Exchange-to-IMAP/SMTP gateway. See the solution in #601.

**Comment by kuochang** [comment]
Thanks @SentToDevNull but my situation might be a bit different since isync has downloaded all O365 emails into my local maildir with no problem.(BTW, my daughter school O365 account might face #601 issue but my office O365 account(my emails from this account) seems works fine with isync and mstmps...)

Those emails that cannot be displayed in neomutt through "notmuch search" are actually displayed normally in neomutt.:) I first thought if Luke's notmuch script rules out some notmuch search result. So I dug further by using command line notmuch to see the differences of the email "id" before and after October 26. Below is the example...

**Email after October 26th:**
```
message{ id:002e01d6b0ff$75841db0$608c5910$@calqulate.io depth:0 match:1 excluded:0 filename:/media/pi/MYGAMEZ/mail/charles/INBOX/cur/1609532878.10387_1277.raspberrypi,U=59107:2,S
header{
<petteri.sutela@calqulate.io> (November 02) (inbox)
... 
```                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                

**Email before October 26th:**                                                                                                                                                                                                                                                                                                                                                  ```
message{ id:HE1PR0502MB361074B089617C7856838E21FB1E0@HE1PR0502MB3610.eurprd05.prod.outlook.com depth:0 match:1 excluded:0 filename:/media/pi/MYGAMEZ/mail/charles/INBOX/cur/1609532839.               10387_1083.raspberrypi,U=58913:2,S
 header{
 Petteri Sutela <petteri.sutela@calqulate.io> (October 19) (attachment inbox)
 ...
```
 
You can find out the sender's email address is the same but the "id" changed. The weird thing is if I do "notmuch show {id:xxx} for the first one(email after October 26th) in the command line, it shows _nothing_. Please note the mail file does exist in maildir.

In summary:
1. Notmuch search result in command line works totally fine. But when I use notmuch in neomutt(press ctl-f), some results are not displayed. Those emails that _can not displayed in neomutt through notmuch search_ display normally in neomutt before using notmuch search.
2.  When I use notmuch show _in command line_ for the "id" that can't be displayed in (1), it shows nothing. But the mail files do exist in the maildir.
3. Mail "id" is for the same sender changed after Oct 26th for somehow. I don't know what does it to do with the whole thing...

It is weird, isn't?
- Charles

**Comment by f380cedric** [comment]
Can you try with this macro instead?

```
macro index \Cf "<enter-command>unset wait_key<enter><shell-escape>read -p 'Enter a search term to find with notmuch: ' x; echo \$x >~/.cache/mutt_terms<enter><limit>=i \`notmuch search --limit=600 --output=messages --format=sexp \$(cat ~/.cache/mutt_terms) | perl -le '@a=<>;chomp@a;print substr(join(\"|=i\",@a),1,-1)'
```

I failed to report upstream because I was unsure but IMO, in the original macro `~i EXPR` is used. `EXPR` is interpreted as a regex, and since Message-ID can use char that are special to regex, it may fail (e.g. there is a `$` in your `id:`)

**Comment by kuochang** [comment]
Sorry for belated response since I was in the vacation. I'll try it and let you know you if it works. Thank you!

**Comment by andrewzigerelli** [comment]
Even though this is closed, I replaced the current macro with the one by @f380cedric.

I was having an issue that if I searched for a certain sender using notmuch in neomutt, I got no messages returned. But searching in the shell, I got the correct result. I just ran the subshell portion in the macro, fixing escapes as necessary, since it caches the last search in ~/.cache/mutt_terms. 

I'm not yet sure why the current macro doesn't work for certain search terms. I wasn't using a complicated expression, just `from:/chars/ `
